### DISCUSSION TITLE
How to store access_token in memory instead of Local Storage in AuthProvider

### DISCUSSION THREAD

**Discussion by heriparid** [refinedev/refine#6031] [General]
I want to store access_token given by Backend to be put in memory instead of Localstorage for security recommendation.
Is there a way to put the token like "global" state so that the token can be fetched using hook.

So far I just found the example using localstorage.
Kindly to hear some recommendation and example of handling the token put in global state.
Thank you.

**Comment by BatuhanW** [comment]
Hey @heriparid, you can store your user's token with cookie or can create a React provider to keep token. Then you can pass this token to Refine's auth provider initialization. Then in Auth Provider, you can use this value in `check` method.

**Reply by heriparid** [reply]
Hi @BatuhanW ,
I would prefer to explore how to integrate a React Provider in refine project setup. Frankly speaking I'm still a beginner, would you like to show me the example on how to use "React Provider" in AuthProvider.
Thanks for the insight.

**Reply by BatuhanW** [reply]
Hey @heriparid you can join our discord community, we have an AI bot that can help you with questions. 🚀 

https://discord.gg/refine
